From khali@linux-fr.org  Mon May 19 10:20:33 2008
From: Jean Delvare <khali@linux-fr.org>
Date: Mon, 19 May 2008 11:14:02 +0200
Subject: i2c/max6875: Really prevent 24RF08 corruption
To: stable@kernel.org
Cc: Ben Gardner <bgardner@wabtec.com>
Message-ID: <20080519111402.4aec493b@hyperion.delvare>

From: Jean Delvare <khali@linux-fr.org>

commit 70455e790391dac85d9b483a9e286a40df1ecc7f in upstream

i2c-core takes care of the possible corruption of 24RF08 chips for
quite some times, so device drivers no longer need to do it. And they
really should not, as applying the prevention twice voids it.

I thought that I had fixed all drivers long ago but apparently I had
missed that one.

Signed-off-by: Jean Delvare <khali@linux-fr.org>
Cc: Ben Gardner <bgardner@wabtec.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/i2c/chips/max6875.c |    3 ---
 1 file changed, 3 deletions(-)

--- a/drivers/i2c/chips/max6875.c
+++ b/drivers/i2c/chips/max6875.c
@@ -207,9 +207,6 @@ static int max6875_detect(struct i2c_ada
 	fake_client->flags = 0;
 	strlcpy(fake_client->name, "max6875 subclient", I2C_NAME_SIZE);
 
-	/* Prevent 24RF08 corruption (in case of user error) */
-	i2c_smbus_write_quick(real_client, 0);
-
 	if ((err = i2c_attach_client(real_client)) != 0)
 		goto exit_kfree2;
 
